doc/overview.md

```diff
@@ -0,0 +1,67 @@
+# Overview
+
+As enterprises migrate from traditional VM infrastructures to more agile
+microservice-based deployments on container platforms, there is a need for
+a simplifying technology that can take care of common cross-cutting
+capabilities around service communication and management such as secure
+interconnect, service discovery & load balancing, staged rollouts, A/B
+testing, intelligent rate limiting, authentication, access control,
+monitoring, logging, etc.
+
+Broadly speaking, an _open, platform independent service mesh_ that
+simplifies traffic management, policy enforcement, and telemetry collection
+has two main benefits. It allows application developers to focus on the
+business logic and iterate quickly on new features by managing how traffic
+flows across their services. It simplifies the operators job of enforcing
+various policies and monitor the mesh from a central control point,
+independent of the evolution of the application ensuring continuous
+compliance with policies of the organization/business unit.
+
+## Architecture
+
+The Istio service mesh consists of three major components:
+
+- **Proxy**. The Istio proxy is designed to mediate inbound and outbound
+traffic for all Istio-managed services. The Istio proxy is based on
+[Envoy](https://lyft.github.io/envoy/) with extensions to fit within the
+Istio service mesh. Envoy has been operational in production at Lyft at
```

Comment:
Maybe get rid of the marketing blurb for Envoy?

Maybe join the next paragraph so the three bullet points read better?